### ISSUE TITLE
Unused object generates invalid JS code

### ISSUE THREAD

**Issue by csakatoku** [zynga/jasy#9]
Unused object which has a function in its properties generate invalid JS code.

The  following code

```
core.Class("p.MyClass", {
    construct: function() {
        var obj = {
            method: function() {
                return "result";
            }
        };
   }
});
```

is compressed to the following code (beautified to make it readable).

```
core.Class("p.MyClass", {
    construct: function () {
        {
            method: function () {
                return "result"
            }
        }
    }
});
```

And this code causes an error saying "Uncaught SyntaxError: Unexpected token (".

I'm using the latest Jasy (f86953ef31), but it's reproducible on Jasy==0.4.6.

Note that the compressed output code is generated by storeCompressed function with the default optimazation options.

```
compressedCode = storeCompressed("build/app-%s.js" % permutation.getChecksum(), classes,
        permutation=permutation, bootCode="new p.MyClass();")
```

**Comment by swernerx** [comment]
Okay, interesting. Might be the unused optimizer which leads to that problem. Will have a look at it. Thanks for reporting.

**Comment by swernerx** [comment]
Verified. Will analyze it now.

**Comment by swernerx** [comment]
Fixed with 376ba9239a

**Comment by swernerx** [comment]
Closing. Thanks for the report!
